sklearn/utils/extmath.py

```diff
@@ -322,6 +321,11 @@ def randomized_svd(M, n_components, n_oversamples=10, n_iter='auto',
         # this implementation is a bit faster with smaller shape[1]
         M = M.T
 
+    if isinstance(M, (sparse.lil_matrix, sparse.dok_matrix)):
+        warnings.warn("Calculating SVD of a {} is expensive. "
+                      "csr_matrix is more efficient.".format(
+                          type(M).__name__))
```

Comment:
Actually adding a unit test under `sklearn/utils/tests/test_extmath.py` to ensure this warning is raised would be useful.  Also in earlier version this raised `SparseEfficiencyWarning` I didn't mean to suggest to remove that.

Reply:
You caught me right in the middle of writing the test! Also my apologies for dropping off the warning type, that's my poor attention span.